plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,273 @@
+
+            autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
+        } catch (RuntimeException | IOException e) {
+            reTryTimes++;
+            // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+            saveLatestReTryTimes(nodeId, reTryTimes);
+            log.error("Can't auto reload model in node id {} ,has try {} times\nThe reason is:{}", nodeId, reTryTimes, e.getMessage());
+        }
+
+        // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+        saveLatestReTryTimes(nodeId, reTryTimes);
+    }
+
+    /**
+     * query task index, and get the result of "task_type"="LOAD_MODEL" and "state"="COMPLETED"
+     * and "worker_node" match nodeId
+     * @param nodeId one of query condition
+     * @return SearchResponse
+     */
+    @VisibleForTesting
+    SearchResponse queryTask(String nodeId) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().from(0).size(1);
+
+        QueryBuilder queryBuilder = QueryBuilders
+            .boolQuery()
+            .must(QueryBuilders.matchPhraseQuery("task_type", "LOAD_MODEL"))
+            .must(QueryBuilders.matchPhraseQuery("state", "COMPLETED"))
+            .must(QueryBuilders.matchPhraseQuery("worker_node", nodeId));
+        searchSourceBuilder.query(queryBuilder);
+
+        SortBuilder sortBuilderOrder = new FieldSortBuilder("last_update_time").order(SortOrder.DESC);
+        searchSourceBuilder.sort(sortBuilderOrder);
+
+        SearchRequest searchRequest = new SearchRequest(ML_TASK_INDEX);
+        searchRequest.source(searchSourceBuilder).indices(ML_TASK_INDEX);
+        try {
+            return client.execute(SearchAction.INSTANCE, searchRequest).actionGet(5000);
+        } catch (IndexNotFoundException e) {
+            throw new IndexNotFoundException("index " + ML_TASK_INDEX + " not found");
+        }
+    }
+
+    /**
+     *  auto reload 1 model under the node id
+     * @param nodeId node id
+     * @param modelId model id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeAndModelId(String nodeId, String modelId) {
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, new String[] { nodeId }, false, false);
+        client.execute(MLLoadModelAction.INSTANCE, mlLoadModelRequest).actionGet(5000);
+    }
+
+    /**
+     * get retry times from the index ".plugins-ml-model-reload" by 1 ml node
+     * @param nodeId the filter condition to query
+     * @return retry times
+     */
+    @VisibleForTesting
+    int getReTryTimes(String nodeId) {
+        if (!isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
+            return 0;
+        }
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MODEL_LOAD_RETRY_TIMES_FIELD }, null);
+        QueryBuilder queryBuilder = QueryBuilders.idsQuery().addIds(nodeId);
```

Comment:
We shouldn't use nodeId, a possible solution is to use nodeId + modelId as the document id.

Reply:
node level auto reload no need model id, resolving this.